src/flint/types/fmpz_mod_poly.pyx

```diff
@@ -0,0 +1,1297 @@
+        fmpz_mod_poly_ctx(170141183460469231731687303715884105727)
+
+    """
+    def __cinit__(self):
+        pass
+
+    def __dealloc__(self):
+        pass
+
+    def __init__(self, mod):
+        # Allow context to be made from fmpz_mod_ctx
+        if typecheck(mod, fmpz_mod_ctx):
+            self.mod = mod
+        else: # Otherwise attempt to create context from moduli
+            self.mod = fmpz_mod_ctx(mod)
+
+    def modulus(self):
+        """
+        Return the modulus from the context as an ``fmpz``
+        type
+
+            >>> R = fmpz_mod_poly_ctx(2**127 - 1)
+            >>> R.modulus()
+            170141183460469231731687303715884105727
+
+        """
+        return self.mod.modulus()
+
+    def is_prime(self):
+        """
+        Return whether the modulus is prime
+
+            >>> fmpz_mod_poly_ctx(2**127).is_prime()
+            False
+            >>> fmpz_mod_poly_ctx(2**127 - 1).is_prime()
+            True
+        """
+        return self.mod.is_prime()
+
+    def zero(self):
+        """
+        Return the zero element of this polynomial ring
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> R.zero()
+            0
+        """
+        cdef fmpz_mod_poly res
+        res = fmpz_mod_poly.__new__(fmpz_mod_poly)
+        res.ctx = self
+        fmpz_mod_poly_zero(res.val, self.mod.val)
+        return res
+
+    def one(self):
+        """
+        Return the one element of this polynomial ring
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> R.one()
+            1
+        """
+        cdef fmpz_mod_poly res
+        res = fmpz_mod_poly.__new__(fmpz_mod_poly)
+        res.ctx = self
+        fmpz_mod_poly_one(res.val, self.mod.val)
+        return res
+
+    def gen(self):
+        """
+        Return the generator of the polynomial: `x`
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> R.gen()
+            x
+        """
+        cdef fmpz_mod_poly res
+        res = fmpz_mod_poly.__new__(fmpz_mod_poly)
+        res.ctx = self
+        fmpz_mod_poly_set_coeff_ui(res.val, 1, 1, self.mod.val)
+        return res
+
+    def random_element(self, degree=3, monic=False, irreducible=False):
+        """
+        Return a random element of degree ``degree``. If ``monic`` is ``True``,
+        ensures the output is monic. If ``irreducible`` is ``True``, ensures
+        that the output is irreducible.
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> f = R.random_element()
+            >>> f.degree()
+            3
+            >>> f = R.random_element(degree=123)
+            >>> f.degree()
+            123
+            >>> f = R.random_element(monic=True)
+            >>> f.is_monic()
+            True
+            >>> f = R.random_element(degree=13, monic=True, irreducible=True)
+            >>> f.degree()
+            13
+            >>> f.is_monic()
+            True
+            >>> f.is_irreducible()
+            True
```

Comment:
I think these doc examples would be clearer if they printed out the polynomials (perhaps reduce the degree from 123) rather than showing things `f.is_monic()`. The degree and the fact that the polynomial is monic should be obvious just looking at the polynomials although irreducibility is less obvious so showing is_irreducible makes sense.

Reply:
I would need to set the random seed in the docstring for the printed output to match with docstring tests... Is this something we can do currently in python-flint?

Reply:
Ideally some way to initialise the random state deterministically should be added. I don't see any Flint function for that though...

In any case for now we can just use `doctest: +SKIP` to skip any parts that would otherwise fail because of different random outputs.